packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,428 @@
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticRecommendationList && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticRecommendationList) {
+      element.setInitializationError();
+    }
+  };
+}
+
+describe('c-quantic-recommendation-list', () => {
+  beforeEach(() => {
+    mockSuccessfulHeadlessInitialization();
+    prepareHeadlessState();
+    console.error = jest.fn();
+  });
+
+  afterEach(() => {
+    cleanup();
+    isInitialized = false;
+    recommendationListState = initialRecommendationListState;
+  });
+
+  describe('when the component is loading', () => {
+    it('should display the placeholder component when loading', async () => {
+      recommendationListState = {
+        isLoading: true,
+        recommendations: [],
+        error: null,
+        searchResponseId: null,
+      };
+      const element = createTestComponent();
+      await flushPromises();
+
+      const loadingPlaceholder = element.shadowRoot.querySelector(
+        selectors.placeholder
+      );
+
+      expect(loadingPlaceholder).not.toBeNull();
+    });
+  });
+
+  describe('controller initialization', () => {
+    it('should build the buildRecommendationList controller with the proper parameters and subscribe to its state', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.buildRecommendationList).toHaveBeenCalledTimes(1);
+      expect(functionMocks.buildRecommendationList).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          options: {
+            id: defaultOptions.recommendation,
+            numberOfRecommendations: Number(
+              defaultOptions.numberOfRecommendations
+            ),
+          },
+        }
+      );
+    });
+
+    it('should build the buildResultTemplatesManager controller with the proper parameters', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.buildResultTemplatesManager).toHaveBeenCalledTimes(
+        1
+      );
+      expect(functionMocks.buildResultTemplatesManager).toHaveBeenCalledWith(
+        exampleEngine
+      );
+    });
+
+    it('should dispatch quantic__registerrecommendationtemplates event', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const event = new CustomEvent(
+        'quantic__registerrecommendationtemplates',
+        {
+          bubbles: true,
+          detail: {
+            content: defaultOptions.recommendation,
+            conditions: [],
+          },
+        }
+      );
+      element.shadowRoot.dispatchEvent(event);
```

Comment:
The component should dispatch this custom event not the test 🤨

Reply:
fixed that